pkg/statistics/handle/util/table_info.go

```diff
@@ -59,11 +61,12 @@ func (c *tableInfoGetterImpl) TableInfoByID(is infoschema.InfoSchema, physicalID
 
 func buildPartitionID2TableID(is infoschema.InfoSchema) map[int64]int64 {
 	mapper := make(map[int64]int64)
-	for _, dbName := range is.AllSchemaNames() {
-		tbls := is.SchemaTableInfos(dbName)
-		for _, tbl := range tbls {
+	rs := is.ListTablesWithSpecialAttribute(infoschema.PartitionAttribute)
+	for _, db := range rs {
+		for _, tbl := range db.TableInfos {
 			pi := tbl.GetPartitionInfo()
 			if pi == nil {
+				logutil.BgLogger().Error("table with partition attribute doesn't have partition info", zap.String("table", tbl.Name.O))
 				continue
 			}
```

Comment:
This seems to be dead code.. Why not remove this?

Reply:
I don't want to use panic here.